extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -85,6 +88,9 @@ InfinispanPropertiesBuildItem setup(ApplicationArchivesBuildItem applicationArch
 
         feature.produce(new FeatureBuildItem(Feature.INFINISPAN_CLIENT));
         additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(InfinispanClientProducer.class));
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(CacheInvalidateAllInterceptor.class));
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(CacheResultInterceptor.class));
+        additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(CacheInvalidateInterceptor.class));
```

Comment:
yes. if I don't add them the interceptors are not discoverable and instanced. I don't use the beans.xml file in the extension, since some beans are optional (health checks for example) and needed only when we use the health check extension and the infinispan client extension at the same time.